src/plugins/imagery/components/ImageryViewLayout.vue

```diff
@@ -207,121 +311,202 @@ export default {
                 });
+                this.setFocusedImage(values[values.length - 1]);
+            }
         },
         timeSystemChange(system) {
-            // reset timesystem dependent variables
-            this.timeKey = system.key;
-            this.timeFormat = this.openmct.telemetry.getValueFormatter(this.metadata.value(this.timeKey));
+            this.timeSystem = this.openmct.time.timeSystem();
+            this.timeKey = this.timeSystem.key;
+            this.timeFormatter = this.getFormatter(this.timeKey);
+            this.durationFormatter = this.getFormatter(this.timeSystem.durationFormat || DEFAULT_DURATION_FORMATTER);
+            this.trackDuration();
+        },
+        clockChange(clock) {
+            this.trackDuration();
         },
         subscribe() {
             this.unsubscribe = this.openmct.telemetry
                 .subscribe(this.domainObject, (datum) => {
-                    let parsedTimestamp = this.timeFormat.parse(datum);
+                    let parsedTimestamp = this.parseTime(datum);
+                    let bounds = this.openmct.time.bounds();
 
-                    if (parsedTimestamp >= this.bounds.start && parsedTimestamp <= this.bounds.end) {
+                    if (parsedTimestamp >= bounds.start && parsedTimestamp <= bounds.end) {
                         this.updateHistory(datum);
+                        this.setFocusedImage(datum);
                     }
                 });
         },
-        unselectAllImages() {
-            this.imageHistory.forEach(image => image.selected = false);
-        },
-        updateHistory(datum, updateValues = true) {
+        updateHistory(datum, setFocused = true) {
             if (this.datumIsNotValid(datum)) {
                 return;
             }
 
             this.imageHistory.push(datum);
 
-            if (updateValues) {
-                this.updateValues(datum);
+            if (setFocused) {
+                this.setFocusedImage(datum);
             }
         },
-        updateValues(datum) {
-            if (this.isPaused) {
-                this.nextDatum = datum;
+        getFormatter(key) {
+            let metadataValue = this.metadata.value(key) || { format: key };
+            let valueFormatter = this.openmct.telemetry.getValueFormatter(metadataValue);
 
-                return;
+            return valueFormatter;
+        },
+        trackDuration() {
+            if (this.canTrackDuration) {
+                this.stopDurationTracking();
+                this.setFormattedDuration();
+                this.durationTracker = window.setInterval(
+                    this.setFormattedDuration, DURATION_TRACK_MS
+                );
+            } else {
+                this.stopDurationTracking();
             }
+        },
+        stopDurationTracking() {
+            window.clearInterval(this.durationTracker);
+            this.formattedDuration = undefined;
+        },
+        getNumericDuration() {
```

Comment:
I meant for this to be calculated once, when the `setInterval` callback is invoked. No need to keep re-evaluating it. Then the formatted duration can just be a computed property that is only re-evaluated when `setInterval` is invoked.

Reply:
Ok... I think I got this one now. I'm only updating the numericDuration in the tracking interval, then the formattedDuration is now a computed that updates when numericDuration is updated.